### DISCUSSION TITLE
Why my parcel app along with babel does not work in ie11

### DISCUSSION THREAD

**Discussion by VelynnXV** [babel/babel#12823] [Q&A]
Hello,

This is my first time using Babel, I've made a very simple app and i tried to make it work on an old browser like ie11, I did that by using babel. Forw some reasons i ignore, my JS does not work at all in ie11 even tho my js files in my dist folder were adapted by babel for ie11 (let variables were replaced by var etc..).

I have no idea what I am missing & I am not sure if my configuration is correct.

My source code :https://github.com/VelynnXV/Front-End-Workflow

the website: https://nifty-noether-cafbd5.netlify.app/

.babelrc
```json
{
  "presets": [
    ["@parcel/babel-preset-env", {
      "targets": {
        "ie": "11"
      }
    }]
  ],
  "plugins": ["@parcel/babel-plugin-transform-runtime"]
}
```

package.json
```json
{
  "name": "mypackage",
  "version": "1.0.0",
  "description": "This is my package, using Parcel. Parcel is a web application bundler, differentiated by its developer experience. It offers blazing fast performance utilizing multicore processing, and requires zero configuration.",
  "app": "public/index.html",
  "scripts": {
    "dev": "npm run clean && parcel public/index.html --dist-dir development -p 3000",
    "build": "rimraf ./dist && parcel build public/*.html --dist-dir dist --public-url ./ --no-optimize",
    "clean": "rimraf ./development && rimraf ./.parcel-cache"
  },
  "browserslist": [
    "IE 11",
    "> 1%"
  ],
  "author": "velynn",
  "license": "UNLICENSED",
  "devDependencies": {
    "@parcel/babel-plugin-transform-runtime": "^2.0.0-nightly.1823",
    "@parcel/babel-preset-env": "^2.0.0-alpha.3",
    "@parcel/optimizer-cssnano": "^2.0.0-nightly.598",
    "@parcel/optimizer-htmlnano": "^2.0.0-nightly.598",
    "@parcel/packager-css": "^2.0.0-nightly.598",
    "@parcel/packager-html": "^2.0.0-nightly.598",
    "@parcel/transformer-css": "^2.0.0-nightly.598",
    "@parcel/transformer-html": "^2.0.0-nightly.598",
    "@parcel/transformer-inline-string": "^2.0.0-nightly.598",
    "@parcel/transformer-postcss": "^2.0.0-nightly.598",
    "@parcel/transformer-posthtml": "^2.0.0-nightly.598",
    "@parcel/transformer-sass": "^2.0.0-nightly.598",
    "@webcomponents/webcomponentsjs": "^2.5.0",
    "parcel": "^2.0.0-beta.1",
    "postcss": "^8.2.6",
    "sass": "^1.32.7"
  },
  "dependencies": {}
}

```

**Comment by JLHwung** [comment]
Can you attach the error stacks on IE 11?

**Reply by VelynnXV** [reply]
Hey! I am on a macOs and i used a simulator for ie11 on the internet like this one https://ieonchrome.com/. I can't find any error msg, all i've noticed is that my JS does not run. I dont have a virtual Machine installed :(

**Comment by nicolo-ribaudo** [comment]
One possible problem is that you are using Babel to compile syntax, but you also need a polyfill (such as `core-js-bundle` or `es5-shim`+`es6-shim`).

**Reply by VelynnXV** [reply]
@nicolo-ribaudo hello, thanks for the reply. So how can include `core-js-bundle` and make it work with everything else? I have some difficulty to visualize how all of those things work together. For example, what role will core-js-bundle play in the compilation?


I modified my .babelrc to this, but i'm not sure of what i'm doing.
```
{
  "presets": [
    ["@parcel/babel-preset-env", {
      "targets": {
        "ie": "11"
      },
      "useBuiltIns": "usage",
      "corejs": 3
    }]
  ],
  "plugins": ["@parcel/babel-plugin-transform-runtime"]
}
```

**Reply by nicolo-ribaudo** [reply]
This is one way of making it work (assuming that you all compiling all your files, including dependencies).
Otherwise you can include `core-js-bundle` with a `<script>` tag in your HTML.

**Reply by VelynnXV** [reply]
I have imported `'core-js/stable'` in my app.js. It still does not work :(

```js
import './scss/app.scss'
import 'core-js/stable'

// import animeCard from './components/animeCard.js'
import trending from './components/trending.js'

console.log('app Entry point')

const app = async () =>{
    document.getElementById('trending-anime').innerHTML= await trending()
    // document.getElementById('cards-container').innerHTML= await animeCard()
}

//Init app
app()
```

**Reply by nicolo-ribaudo** [reply]
Then we'll need to know if it's throwing an error, otherwise it's impossible for us to help :confused:

**Reply by VelynnXV** [reply]
@nicolo-ribaudo i just noticed my build is failing in netlify.com : `4:36:26 PM: @parcel/core: Failed to resolve 'core-js/modules/es.symbol.js' from './src/app.js'
`
'

> 
> 4:36:03 PM: Build ready to start
> 4:36:05 PM: build-image version: d84c79427e8f83c1ba17bcdd7b3fe38059376b68
> 4:36:05 PM: build-image tag: v3.6.1
> 4:36:05 PM: buildbot version: df42d44cc6a0381cf657a671baf049c03981bafe
> 4:36:05 PM: Fetching cached dependencies
> 4:36:06 PM: Starting to download cache of 106.5MB
> 4:36:06 PM: Finished downloading cache in 686.197206ms
> 4:36:06 PM: Starting to extract cache
> 4:36:11 PM: Finished extracting cache in 4.714555558s
> 4:36:11 PM: Finished fetching cache in 5.444777996s
> 4:36:11 PM: Starting to prepare the repo for build
> 4:36:11 PM: Preparing Git Reference refs/heads/main
> 4:36:13 PM: Different publish path detected, going to use the one specified in the Netlify configuration file: 'dist' versus 'dist/' in the Netlify UI
> 4:36:13 PM: Starting build script
> 4:36:13 PM: Installing dependencies
> 4:36:13 PM: Python version set to 2.7
> 4:36:14 PM: Started restoring cached node version
> 4:36:17 PM: Finished restoring cached node version
> 4:36:17 PM: v12.18.0 is already installed.
> 4:36:18 PM: Now using node v12.18.0 (npm v6.14.4)
> 4:36:18 PM: Started restoring cached build plugins
> 4:36:18 PM: Finished restoring cached build plugins
> 4:36:18 PM: Attempting ruby version 2.7.1, read from environment
> 4:36:20 PM: Using ruby version 2.7.1
> 4:36:20 PM: Using PHP version 5.6
> 4:36:20 PM: Started restoring cached node modules
> 4:36:20 PM: Finished restoring cached node modules
> 4:36:20 PM: Started restoring cached go cache
> 4:36:21 PM: Finished restoring cached go cache
> 4:36:21 PM: go version go1.14.4 linux/amd64
> 4:36:21 PM: go version go1.14.4 linux/amd64
> 4:36:21 PM: Installing missing commands
> 4:36:21 PM: Verify run directory
> 4:36:23 PM: ​
> 4:36:23 PM: ────────────────────────────────────────────────────────────────
> 4:36:23 PM:   Netlify Build                                                 
> 4:36:23 PM: ────────────────────────────────────────────────────────────────
> 4:36:23 PM: ​
> 4:36:23 PM: ❯ Version
> 4:36:23 PM:   @netlify/build 9.1.3
> 4:36:23 PM: ​
> 4:36:23 PM: ❯ Flags
> 4:36:23 PM:   deployId: 602eddc39842fd0007058639
> 4:36:23 PM:   mode: buildbot
> 4:36:23 PM: ​
> 4:36:23 PM: ❯ Current directory
> 4:36:23 PM:   /opt/build/repo
> 4:36:23 PM: ​
> 4:36:23 PM: ❯ Config file
> 4:36:23 PM:   No config file was defined: using default values.
> 4:36:23 PM: ​
> 4:36:23 PM: ❯ Context
> 4:36:23 PM:   production
> 4:36:23 PM: ​
> 4:36:23 PM: ────────────────────────────────────────────────────────────────
> 4:36:23 PM:   1. Build command from Netlify app                             
> 4:36:23 PM: ────────────────────────────────────────────────────────────────
> 4:36:23 PM: ​
> 4:36:23 PM: $ npm run build
> 4:36:23 PM: > mypackage@1.0.0 build /opt/build/repo
> 4:36:24 PM: > rimraf ./dist && parcel build public/*.html --dist-dir dist --public-url ./ --no-optimizeBuilding...
> 4:36:26 PM: 🚨 Build failed.
> 4:36:26 PM: @parcel/core: Failed to resolve 'core-js/modules/es.symbol.js' from './src/app.js'
> 4:36:26 PM: npm ERR! code ELIFECYCLE
> 4:36:26 PM: npm ERR! errno 1
> 4:36:26 PM: npm ERR! mypackage@1.0.0 build: `rimraf ./dist && parcel build public/*.html --dist-dir dist --public-url ./ --no-optimize`
> 4:36:26 PM: npm ERR! Exit status 1
> 4:36:26 PM: npm ERR!
> 4:36:26 PM: npm ERR! Failed at the mypackage@1.0.0 build script.
> 4:36:26 PM: npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
> 4:36:26 PM: npm ERR! A complete log of this run can be found in:
> 4:36:26 PM: npm ERR!     /opt/buildhome/.npm/_logs/2021-02-18T21_36_26_501Z-debug.log
> 4:36:26 PM: ​
> 4:36:26 PM: ────────────────────────────────────────────────────────────────
> 4:36:26 PM:   "build.command" failed                                        
> 4:36:26 PM: ────────────────────────────────────────────────────────────────
> 4:36:26 PM: ​
> 4:36:26 PM:   Error message
> 4:36:26 PM:   Command failed with exit code 1: npm run build
> 4:36:26 PM: ​
> 4:36:26 PM:   Error location
> 4:36:26 PM:   In Build command from Netlify app:
> 4:36:26 PM:   npm run build
> 4:36:26 PM: ​
> 4:36:26 PM:   Resolved config
> 4:36:26 PM:   build:
> 4:36:26 PM:     command: npm run build
> 4:36:26 PM:     commandOrigin: ui
> 4:36:26 PM:     publish: /opt/build/repo/dist
> 4:36:26 PM: Caching artifacts
> 4:36:26 PM: Started saving node modules
> 4:36:26 PM: Finished saving node modules
> 4:36:26 PM: Started saving build plugins
> 4:36:26 PM: Finished saving build plugins
> 4:36:26 PM: Started saving pip cache
> 4:36:26 PM: Finished saving pip cache
> 4:36:26 PM: Started saving emacs cask dependencies
> 4:36:26 PM: Finished saving emacs cask dependencies
> 4:36:26 PM: Started saving maven dependencies
> 4:36:26 PM: Finished saving maven dependencies
> 4:36:26 PM: Started saving boot dependencies
> 4:36:26 PM: Finished saving boot dependencies
> 4:36:26 PM: Started saving rust rustup cache
> 4:36:26 PM: Finished saving rust rustup cache
> 4:36:26 PM: Started saving go dependencies
> 4:36:26 PM: Finished saving go dependencies
> 4:36:26 PM: Build failed due to a user error: Build script returned non-zero exit code: 2
> 4:36:26 PM: Failing build: Failed to build site
> 4:36:26 PM: Failed during stage 'building site': Build script returned non-zero exit code: 2
> 4:36:26 PM: Finished processing build request in 21.201590564s

Here are my configs

.babelrc
```
{
  "presets": [
    ["@parcel/babel-preset-env", {
      "targets": {
        "ie": "11"
      },
      "useBuiltIns": "entry",
      "corejs": 3
    }]
  ],
  "plugins": ["@parcel/babel-plugin-transform-runtime"]
}
```
package.json
```
{
  "name": "mypackage",
  "version": "1.0.0",
  "description": "This is my package, using Parcel. Parcel is a web application bundler, differentiated by its developer experience. It offers blazing fast performance utilizing multicore processing, and requires zero configuration.",
  "app": "public/index.html",
  "scripts": {
    "dev": "npm run clean && parcel public/index.html --dist-dir development -p 3000",
    "build": "rimraf ./dist && parcel build public/*.html --dist-dir dist --public-url ./ --no-optimize",
    "clean": "rimraf ./development && rimraf ./.parcel-cache"
  },
  "browserslist": [
    "IE 11",
    "> 1%"
  ],
  "author": "velynn",
  "license": "UNLICENSED",
  "devDependencies": {
    "@parcel/babel-plugin-transform-runtime": "^2.0.0-nightly.1823",
    "@parcel/babel-preset-env": "^2.0.0-alpha.3",
    "@parcel/optimizer-cssnano": "^2.0.0-nightly.598",
    "@parcel/optimizer-htmlnano": "^2.0.0-nightly.598",
    "@parcel/packager-css": "^2.0.0-nightly.598",
    "@parcel/packager-html": "^2.0.0-nightly.598",
    "@parcel/transformer-css": "^2.0.0-nightly.598",
    "@parcel/transformer-html": "^2.0.0-nightly.598",
    "@parcel/transformer-inline-string": "^2.0.0-nightly.598",
    "@parcel/transformer-postcss": "^2.0.0-nightly.598",
    "@parcel/transformer-posthtml": "^2.0.0-nightly.598",
    "@parcel/transformer-sass": "^2.0.0-nightly.598",
    "@webcomponents/webcomponentsjs": "^2.5.0",
    "parcel": "^2.0.0-beta.1",
    "postcss": "^8.2.6",
    "sass": "^1.32.7"
  },
  "dependencies": {}
}

```

**Reply by nicolo-ribaudo** [reply]
Run `npm install core-js`

**Reply by VelynnXV** [reply]
@nicolo-ribaudo 
I installed core-js and I added

```js
import 'core-js/stable'
import "regenerator-runtime/runtime";
```
In my app.js

package.json
```json
{
  "name": "mypackage",
  "version": "1.0.0",
  "description": "This is my package, using Parcel. Parcel is a web application bundler, differentiated by its developer experience. It offers blazing fast performance utilizing multicore processing, and requires zero configuration.",
  "app": "public/index.html",
  "scripts": {
    "dev": "npm run clean && parcel public/index.html --dist-dir development -p 3000",
    "build": "rimraf ./dist && parcel build public/*.html --dist-dir dist --public-url ./ --no-optimize",
    "clean": "rimraf ./development && rimraf ./.parcel-cache"
  },
  "browserslist": [
    "IE 11",
    "> 1%"
  ],
  "author": "velynn",
  "license": "UNLICENSED",
  "devDependencies": {
    "@parcel/babel-plugin-transform-runtime": "^2.0.0-nightly.1823",
    "@parcel/babel-preset-env": "^2.0.0-alpha.3",
    "@parcel/optimizer-cssnano": "^2.0.0-nightly.598",
    "@parcel/optimizer-htmlnano": "^2.0.0-nightly.598",
    "@parcel/packager-css": "^2.0.0-nightly.598",
    "@parcel/packager-html": "^2.0.0-nightly.598",
    "@parcel/transformer-css": "^2.0.0-nightly.598",
    "@parcel/transformer-html": "^2.0.0-nightly.598",
    "@parcel/transformer-inline-string": "^2.0.0-nightly.598",
    "@parcel/transformer-postcss": "^2.0.0-nightly.598",
    "@parcel/transformer-posthtml": "^2.0.0-nightly.598",
    "@parcel/transformer-sass": "^2.0.0-nightly.598",
    "@webcomponents/webcomponentsjs": "^2.5.0",
    "core-js": "^3.9.0",
    "parcel": "^2.0.0-beta.1",
    "postcss": "^8.2.6",
    "sass": "^1.32.7"
  },
  "dependencies": {}
}

```


.babelrc
```
{
  "presets": [
    ["@parcel/babel-preset-env", {
      "targets": {
        "ie": "11"
      },
      "useBuiltIns": "entry",
      "corejs": 3.8
    }]
  ],
  "plugins": ["@parcel/babel-plugin-transform-runtime"]
}
```

But when i build and run my app i get : 

> Uncaught ReferenceError: $a9cff48d0a9a1d5e6d3a39756f182d0e$exports is not defined
>     at es.map.js:3

Is there any simple solution to make an app work in ie11? I have never seen a working example of a application made with parcel that works with babel. Maybe that can help me understand how to setup my app.

My source code :https://github.com/VelynnXV/Front-End-Workflow

the website: https://nifty-noether-cafbd5.netlify.app/
